opencsp/app/sofast/lib/SystemSofastFringe.py

```diff
@@ -348,18 +431,53 @@ def capture_mask_and_fringe_images(self, run_next: Callable | None = None) -> No
         """
         # Check fringes/camera have been loaded
         if self.fringes is None:
-            raise ValueError('Fringes have not been loaded.')
+            lt.error_and_raise(
+                ValueError,
+                'Error in SystemSofastFringe.capture_mask_and_fringe_images(): Fringes have not been loaded.',
+            )
 
         def run_after_capture():
             self.capture_fringe_images(run_next)
 
         # Capture mask images, then capture fringe images, then run_next
         self.capture_mask_images(run_after_capture)
 
+    def run_measurement(self, fringes: Fringes, on_done: Callable = None) -> None:

```

Comment:
Would it make things simpler or more complicated if we made `fringes` an attribute of `SystemSofastFringe`? I'm thinking this could be beneficial because declaring the fringe object typically happens one/few times and the call to run the measurement sequence happens often. Might be marginal gains, let me know what you think.

Reply:
I agree with your assertion that 'fringes' should be a property of SystemSofastFringe.  In fact, it already is!  I'm updating fringes to be get-only, and changing the method name to 'set_fringes'.